File: Power Plant Codes/Visual Studio 2019/Web Material Calculator/PowerPlant.Models/ItemMaster.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

#nullable disable

namespace PowerPlant.Models
{
    public partial class ItemMaster
    {
        public string Facility { get; set; }
        public string ItemNumber { get; set; }
        public int ProductionShelfLifeDays { get; set; }
        public decimal LabelWeight { get; set; }
        public string LabelWeightUom { get; set; }
        public string BagLengthRequired { get; set; }
        public decimal BagLength { get; set; }
        public string LabelDateFmtCode { get; set; }
        public decimal PackagesPerSaleableUnit { get; set; }
        public decimal SaleableUnitPerCase { get; set; }
        public int QtyPerPallet { get; set; }
        public string Scccode { get; set; }
        public string Upccode { get; set; }
        public string OverrideItem { get; set; }
        public string ItemDesc1 { get; set; }
        public string ItemDesc2 { get; set; }
        public string ItemDesc3 { get; set; }
        public string PackSize { get; set; }
        public string NetWeight { get; set; }
        public string DomicileText1 { get; set; }
        public string DomicileText2 { get; set; }
        public string DomicileText3 { get; set; }
        public string DomicileText4 { get; set; }
        public string DomicileText5 { get; set; }
        public string DomicileText6 { get; set; }
        public string CaseLabelFmt1 { get; set; }
        public string CaseLabelFmt2 { get; set; }
        public string CaseLabelFmt3 { get; set; }
        public string PackageCoderFmt1 { get; set; }
        public string PackageCoderFmt2 { get; set; }
        public string PackageCoderFmt3 { get; set; }
        public string FilterCoderFmt { get; set; }
        public string ProductionDateDesc { get; set; }
        public string ExpiryDateDesc { get; set; }
        public string PrintSolot { get; set; }
        public string DateToPrintFlag { get; set; }
        public string PrintCaseLabel { get; set; }
        public int Tie { get; set; }
        public int Tier { get; set; }
        public int? ShipShelfLifeDays { get; set; }
        public string ItemType { get; set; }
        public string ItemMajorClass { get; set; }
        public string PalletCode { get; set; }
        public bool? SlipSheet { get; set; }
        public string PkgLabelDateFmtCode { get; set; }
        public decimal StdCostPerLb { get; set; }
        public decimal? GrsDepth { get; set; }
        public decimal? GrsHeight { get; set; }
        public decimal? GrsWidth { get; set; }
        public byte? CaseLabelApplicator { get; set; }
        public bool? InsertBrewerFilter { get; set; }


        public WebMaterial WebMaterial { get; set; }
    }
}

File: Power Plant Codes/Visual Studio 2012/Projects/Package Line Rate Maintenance/Package Line Rate Maintenance/Data/ItemMaster.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity.Spatial

<Table("tblItemMaster")>
Partial Public Class ItemMaster
    <Required>
    <StringLength(3)>
    Public Property Facility As String

    <Key>
    <StringLength(35)>
    Public Property ItemNumber As String

    Public Property ProductionShelfLifeDays As Integer

    Public Property LabelWeight As Decimal

    <Required>
    <StringLength(2)>
    Public Property LabelWeightUOM As String

    <Required>
    <StringLength(1)>
    Public Property BagLengthRequired As String

    Public Property BagLength As Decimal

    <Required>
    <StringLength(2)>
    Public Property LabelDateFmtCode As String

    Public Property PackagesPerSaleableUnit As Decimal

    Public Property SaleableUnitPerCase As Decimal

    Public Property QtyPerPallet As Integer

    <Required>
    <StringLength(14)>
    Public Property SCCCode As String

    <Required>
    <StringLength(14)>
    Public Property UPCCode As String

    <Required>
    <StringLength(35)>
    Public Property OverrideItem As String

    <Required>
    <StringLength(50)>
    Public Property ItemDesc1 As String

    <Required>
    <StringLength(50)>
    Public Property ItemDesc2 As String

    <StringLength(50)>
    Public Property ItemDesc3 As String

    <Required>
    <StringLength(12)>
    Public Property PackSize As String

    <Required>
    <StringLength(10)>
    Public Property NetWeight As String

    <Required>
    <StringLength(24)>
    Public Property DomicileText1 As String

    <Required>
    <StringLength(24)>
    Public Property DomicileText2 As String

    <Required>
    <StringLength(24)>
    Public Property DomicileText3 As String

    <Required>
    <StringLength(24)>
    Public Property DomicileText4 As String

    <Required>
    <StringLength(24)>
    Public Property DomicileText5 As String

    <Required>
    <StringLength(24)>
    Public Property DomicileText6 As String

    <Required>
    <StringLength(25)>
    Public Property CaseLabelFmt1 As String

    <Required>
    <StringLength(25)>
    Public Property CaseLabelFmt2 As String

    <Required>
    <StringLength(25)>
    Public Property CaseLabelFmt3 As String

    <Required>
    <StringLength(25)>
    Public Property PackageCoderFmt1 As String

    <Required>
    <StringLength(25)>
    Public Property PackageCoderFmt2 As String

    <Required>
    <StringLength(25)>
    Public Property PackageCoderFmt3 As String

    <Required>
    <StringLength(25)>
    Public Property FilterCoderFmt As String

    <StringLength(30)>
    Public Property ProductionDateDesc As String

    <StringLength(30)>
    Public Property ExpiryDateDesc As String

    <Required>
    <StringLength(1)>
    Public Property PrintSOLot As String

    <Required>
    <StringLength(1)>
    Public Property DateToPrintFlag As String

    <Required>
    <StringLength(1)>
    Public Property PrintCaseLabel As String

    Public Property Tie As Integer

    Public Property Tier As Integer

    Public Property ShipShelfLifeDays As Integer?

    <Required>
    <StringLength(10)>
    Public Property ItemType As String

    <Required>
    <StringLength(10)>
    Public Property ItemMajorClass As String

    <StringLength(1)>
    Public Property PalletCode As String

    Public Property SlipSheet As Boolean?

    <StringLength(2)>
    Public Property PkgLabelDateFmtCode As String

    Public Property StdCostPerLB As Decimal

    Public Property GrsDepth As Decimal?

    Public Property GrsHeight As Decimal?

    Public Property GrsWidth As Decimal?

    Public Property CaseLabelApplicator As Byte?

    Public Property InsertBrewerFilter As Boolean?
End Class
